pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -15,23 +15,30 @@
 package flowrecords
 
 import (
+	"container/heap"
 	"fmt"
 	"sync"
 	"time"
 
-	"k8s.io/klog/v2"
-
 	"antrea.io/antrea/pkg/agent/flowexporter"
 )
 
 type FlowRecords struct {
-	recordsMap map[flowexporter.ConnectionKey]flowexporter.FlowRecord
-	mutex      sync.Mutex
+	recordsMap          map[flowexporter.ConnectionKey]*flowexporter.FlowRecord
```

Comment:
We cannot remove the map because we need to do map searching at [here](https://github.com/antrea-io/antrea/blob/9069471728e9b4d49f7ec96408163be92fbbb8e2/pkg/agent/flowexporter/flowrecords/flow_records.go#L62). Any other better solutions other than keeping the map?

Reply:
We have the connection store to search right. Can we use that? We add and update items in the priority queue as part of the `AddOrUpdateConnection` method in connections_conntrack.go.
Do you think that can work?

Reply:
I think that's a good idea. I cannot see any concern about this approach currently. Let me work on it.

Reply:
@srikartati Do you see any concern if we remove the whole `FlowRecord` and `FlowRecords` struct? I just realized if we remove the record map then there's no need to keep those two structs. 
Say we'll move the statistics metadata `PrevPackets`, `PrevBytes`, etc. to connection struct, and we update those metadata after every export. For example, in exporter.go - method [`addRecordToSet`](https://github.com/antrea-io/antrea/blob/9069471728e9b4d49f7ec96408163be92fbbb8e2/pkg/agent/flowexporter/exporter/exporter.go#L436), originally we are passing `record.Conn`, which is a connection value. Now we would pass `conn *flowexporter.Connection` as a pointer. Do you think that still works fine, since we fix the `PrevPackets`, `PrevBytes` from last export?

Reply:
Yes removing those structs is fine; that is what I was suggesting [here](https://github.com/antrea-io/antrea/pull/2360#discussion_r671544127).

It is better to keep the stats metadata in priority queue item, so that we do not have to access connection from the exporter goroutine. In addition, let's keep the copy of the connection in the priority queue rather than a pointer.
Let's make sure that only polling goroutine accesses the connection store and exporter goroutine has to access priority queue. We will not be efficient and we may run into deadlock issues if goroutines access both the resources.